### PR TITLE
Remove click and associated CLI functions 

### LINKED ISSUE #2350: Remove click and its associated functions
- We can reduce our package size by removing click, which isn't really that used in Featuretools

### PR DESCRIPTION
**Pull request by gsheni**
- Fixes #2350

### CODE REVIEW

**Review by rwedge — COMMENTED** [review]
`featuretools/tests/cli_tests` folder should be deleted

File: pyproject.toml [hunk 1]
```diff
@@ -108,6 +108,7 @@ docs = [
     "myst-parser == 0.18.0",
     "nlp_primitives >= 2.3.0",
     "autonormalize >= 2.0.1",
+    "click >=7.0.0",
```
Comment (gsheni) [inline pyproject.toml:111]: need for https://github.com/alteryx/featuretools/blob/main/docs/notebook_version_standardizer.py

File: featuretools/utils/utils_info.py [hunk 2]
```diff
@@ -17,15 +16,13 @@
...
-    "Click",
     "pip",
     "setuptools",
 ]
 
 
 def show_info():
-    res = subprocess.run(["featuretools", "info"], stdout=subprocess.PIPE)
-    print(res.stdout.decode("utf-8"))
+    print_info()
```
Comment (rwedge) [inline featuretools/utils/utils_info.py:25]: I think we should put the contents of `print_info` here and delete `print_info`, it's not used anywhere else

Comment (gsheni) [reply featuretools/utils/utils_info.py]: Ah gotcha, fixed.

**Review by rwedge — COMMENTED** [review]
We should remove the CLI entry in pyproject.toml

```
[project.scripts]
featuretools = "featuretools.__main__:cli"
```

File: docs/source/release_notes.rst [hunk 3]
```diff
@@ -18,6 +19,10 @@ Future Release
     Thanks to the following people for contributing to this release:
     :user:`gsheni`, :user:`rwedge`, :user:`sbadithe`, :user:`tamargrey`
 
+Breaking Changes
+++++++++++++++++
+* The CLI functions ``featuretools list-primitives`` and ``featuretools info`` have been removed.
```
[the remark this answers is not in the document]
Comment (gsheni) [reply docs/source/release_notes.rst]: Fixed.

File: featuretools/tests/utils_tests/test_utils_info.py [hunk 4]
```diff
@@ -14,6 +18,17 @@ def this_dir():
     return os.path.dirname(os.path.abspath(__file__))
 
 
+def test_show_info():
+    f = io.StringIO()
+    with redirect_stdout(f):
```
Comment (rwedge) [inline featuretools/tests/utils_tests/test_utils_info.py:23]: pytest has a [capsys fixture](https://docs.pytest.org/en/latest/how-to/capture-stdout-stderr.html#accessing-captured-output-from-a-test-function) to capture output in a test, I think it'd be good to use the pytest approach

Comment (gsheni) [reply featuretools/tests/utils_tests/test_utils_info.py]: Woah, did not know about this. Fixed
https://github.com/alteryx/featuretools/pull/2353/commits/3cba6ffb7854dcadbc14a1533042d4878fe2f1da

**Review by rwedge — APPROVED** [review]
LGTM

**Pull request merged**